### PR TITLE
libdicom: add a new recipe

### PR DESCRIPTION
**Pull request by valgur**
Adds libdicom: https://github.com/ImagingDataCommons/libdicom

[Packaging status](https://repology.org/project/libdicom/versions)

### CODE REVIEW

**Comment by valgur** [comment]
Requires #20964.

**Review by jwillikers — APPROVED** [review]

File: recipes/libdicom/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        meson.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        fix_apple_shared_install_name(self)
+
+    def package_info(self):
+        self.cpp_info.libs = ["dicom"]
+        self.cpp_info.set_property("pkg_config_name", "libdicom")
```
Comment (uilianries) [inline recipes/libdicom/all/conanfile.py:83]:
```suggestion
        self.cpp_info.set_property("pkg_config_name", "libdicom")
        if not self.options.shared:
            self.cpp_info.defines = ["DCM_STATIC"]
```

https://github.com/ImagingDataCommons/libdicom/blob/v1.0.5/include/dicom/dicom.h#L11

Comment (valgur) [reply recipes/libdicom/all/conanfile.py:83]: Good catch, thanks!

Comment (franramirez688) [reply recipes/libdicom/all/conanfile.py:83]: @uilianries  @valgur  The `meson.build` is setting it by getting the `default_library` option. Perhaps, it'd be better to add it via MesonToolchain:

```python
        tc = MesonToolchain(self)
        tc.project_options["tests"] = "false"
        if not self.options.shared:
             tc.project_options["default_library"] = "static"
        tc.generate()
```

File: recipes/libdicom/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        if not self.options.shared:
+            self.cpp_info.defines = ["DCM_STATIC"]
```
Comment (franramirez688) [inline recipes/libdicom/all/conanfile.py:85]: Sorry, I just double-checked it, and MesonToolchain is already defining the `default_library` in the built-in options section, so we can remove that part:

```suggestion
```

File: recipes/libdicom/all/conanfile.py (hunk 1 above)
Comment (franramirez688) [reply recipes/libdicom/all/conanfile.py:83]: Follow-up: https://github.com/conan-io/conan-center-index/pull/20962/files#r1392573896

File: recipes/libdicom/all/conanfile.py (hunk 2 above)
Comment (valgur) [reply recipes/libdicom/all/conanfile.py:85]: Sorry, I'm confused. This exported define is not really related to MesonToolchain?

Comment (franramirez688) [reply recipes/libdicom/all/conanfile.py:85]: Sorry, yes, it's related, and it's already defined by the `default_library` in the `conan_meson_xxxx.ini` within the `[built-in options]` section. See https://github.com/ImagingDataCommons/libdicom/blob/ba0e6d6f1fa08d69a4582b38817ca7ec699705e5/meson.build#L62C46-L62C61

Comment (franramirez688) [reply recipes/libdicom/all/conanfile.py:85]: Discussed with @uilianries Keeping it to prevent missing symbols.

**Review by franramirez688 — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**